studioclass-backend/src/graphql/queries/posts.js

```diff
@@ -2,19 +2,77 @@ import { gql } from 'apollo-server-express';
 import Post from '../../models/post';
 
 export const typeDefs = gql`
+  enum OrderEnum {
+    DATE
+  }
+
+  enum Direction {
+    ASC
+    DESC
+  }
+
   type Query {
-    posts: [Post!]!
+    posts(
+      orderBy: OrderEnum = DATE,
+      orderDirection: Direction = ASC,
+      cursor: String,
+      limit: Int = 10
+    ): PostConnection!
     post(id: ID!): Post
   }
 `;
 
 export const resolvers = {
   Query: {
-    posts: async () => {
-      const posts = await Post.find({})
-        .populate({ path: 'user', select: 'id username'})
-        .populate('tags');
-      return posts;
+    posts: async (_root, { orderBy, orderDirection, cursor, limit }) => {
+      console.log({ orderBy, orderDirection, cursor, limit });
+      let orderOn;
+      switch (orderBy) {
+        case 'DATE':
+          orderOn = '_id'; // MongoDB IDs include date/time info
+          break;
+        default: 
+          orderOn = '_id';
+      }
+
+      const orderDir = orderDirection === 'DESC' ? '$lt' : '$gt';
+      const sortDir = orderDirection === 'DESC' ? -1 : 1;
+
+      let posts = [];
+      
+      if (cursor) {
+        posts = await Post.find({ [orderOn]: { [orderDir]: cursor }})
+          .sort({ [orderOn]: sortDir })
+          .limit(limit)
+          .populate({ path: 'user', select: 'id username'})
+          .populate('tags');
+      } else {
+        posts = await Post.find({})
```

Comment:
This kind of control flow can get tricky. Mutability has a tendency to introduce weird bugs. consider a refactor that prioritizing immutablity.